update-sri.js

```diff
@@ -35,24 +33,37 @@ Promise.all(process.argv.slice(3).map(file =>
         // 5 - integrity opening quote: "
         // 6 - old hash: sha384-…
         // 7 - integrity hash algorithm: sha384
-        const cdnRe = /((["'])https?:\/\/cdn\.jsdelivr\.net\/npm\/katex@)[^\/"']+(\/([^"']+)\2(?:\s+integrity=(["'])(([^-]+)-[^"']+)\5)?)/g;
+        // eslint-disable-next-line max-len
+        const cdnRe = /((["'])https?:\/\/cdn\.jsdelivr\.net\/npm\/katex@)[^/"']+(\/([^"']+)\2(?:\s+integrity=(["'])(([^-]+)-[^"']+)\5)?)/g;
         const hashes = {};
         body = body.replace(cdnRe, (m, pre, oq1, post, file, oq2, old, algo) => {
             if (old) {
-                hashes[old] = { file, algo };
+                hashes[old] = {file, algo};
             }
             return pre + version + post;
         });
-        return Promise.all(Object.keys(hashes).map(hash =>
+        let promise = Promise.all(Object.keys(hashes).map(hash =>
             read(hashes[hash].file, null)
             .then(data => {
-                body = body.replace(hash, sriToolbox.generate({
+                const newHash = sriToolbox.generate({
                     algorithms: [hashes[hash].algo],
-                }, data));
+                }, data);
+                body = body.replace(hash, newHash);
+
+                if (version === "check" && hash !== newHash) {
+                    throw new Error("SRI mismatch! " +
+                        "Please run the release script again.");
+                }
             })
-        )).then(() => write(file, body));
+        ));
+
+        if (version !== "check") {
+            promise = promise.then(() => write(file, body));
```

Comment:
Minor nit: this could just return `promise.then(...)` here then `promise` could be a constant.